### PR TITLE
Multiagent pr

### PR DESCRIPTION
**Pull request by eugenevinitsky**
- Makes base_env multiagent compliant
- Adds a few working multiagent examples
- Adds multiagent handling to visualizer_rllib

### CODE REVIEW

**Comment by mtgibson2014** [comment]
@mtgibson2014 Reviewing currently.

File: examples/sumo/bottleneck.py [hunk 1]
```diff
@@ -107,5 +114,17 @@ def bottleneck_example(flow_rate, horizon, render=None):
...
+        exp = bottleneck_example(INFLOW, 1000, render=True)
+        exp.run(5, 1000)
+    else:
+        num_steps = 20
+        inflow_range = np.linspace(1500, 3500, num_steps)
+        capacity = np.zeros(num_steps)
+        for i in range(inflow_range.shape[0]):
+            exp = bottleneck_example(inflow_range[i], 2000, render=False)
+            info_dict = exp.run(5, 2000)
+            capacity[i] = info_dict["mean_outflows"]
```
Comment (mtgibson2014) [inline examples/sumo/bottleneck.py]: Double quotes!

File: examples/sumo/bottleneck.py [hunk 2]
```diff
@@ -107,5 +114,17 @@ def bottleneck_example(flow_rate, horizon, render=None):
 if __name__ == "__main__":
     # import the experiment variable
     # inflow, number of steps, binary
```
Comment (mtgibson2014) [inline examples/sumo/bottleneck.py]: This is a little confusing. What is this for?

File: flow/benchmarks/bottleneck2.py [hunk 3]
```diff
@@ -55,11 +55,11 @@
     "lane_change_duration": 5,
     "max_accel": 3,
     "max_decel": 3,
-    "inflow_range": [1000, 2000]
+    "inflow_range": [1000 * SCALING, 2000 * SCALING]
```
Comment (mtgibson2014) [inline flow/benchmarks/bottleneck2.py]: Double quotes

File: flow/envs/bottleneck_env.py [hunk 4]
```diff
@@ -936,3 +947,268 @@ def reset(self):
...
+            num_ids = len(self.vehicles.get_ids_by_edge(edge))
+            cong_penalty = max(num_ids - 30, 0)/60
+            return {rl_id: self.vehicles.get_speed(rl_id)/10
+                    - cong_penalty for rl_id
+                    in self.vehicles.get_rl_ids()}
+        else:
+            return {}
+
+    def reset(self):
+        add_params = self.env_params.additional_params
```
Comment (mtgibson2014) [inline flow/envs/bottleneck_env.py]: Comments?

**Comment by eugenevinitsky** [comment]
Merging blocked by need for updated aws sumo binaries @AboudyKreidieh 
Thanks for the review @mtgibson2014 !

File: flow/envs/bottleneck_env.py [hunk 5]
```diff
@@ -936,3 +947,268 @@ def reset(self):
...
+        else:
+            return Box(
+                low=-1.5, high=1.0, shape=(1,), dtype=np.float32)
+
+    def get_state(self, rl_actions=None):
+        """See class definition."""
+        # action space is speed and velocity of leading and following
+        # vehicles for all of the avs
+        if self.env_params.additional_params.get('communicate', False):
+            veh_info = {rl_id: np.concatenate((self.state_util(rl_id),
```
[the remark this answers is not in the document]
Comment (eugenevinitsky) [reply flow/envs/bottleneck_env.py]: Actually, going to remove this from the PR. This shouldn't be public yet.

File: flow/benchmarks/bottleneck2.py (hunk 3 above)
Comment (eugenevinitsky) [reply flow/benchmarks/bottleneck2.py]: Good point here!

**Comment by eugenevinitsky** [comment]
Status: tests are passing locally but not yet on Travis.

File: examples/sumo/bottleneck.py (hunk 2 above)
Comment (eugenevinitsky) [reply examples/sumo/bottleneck.py]: Could you clarify which line is confusing?

**Comment by AboudyKreidieh** [comment]
pretty sure this has been merged in another PR

**Pull request closed without merging**